### ISSUE TITLE
Error at bottom of gravity.list: Binary file /etc/pihole/list.preEventHorizon matches

### ISSUE THREAD

**Issue by StarPicard** [pi-hole/pi-hole#1909]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [] The issue I am reporting can be *replicated*.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`4`

---
**Expected behaviour:**

The file `gravity.list` should be a normal host-file.

**Actual behaviour:**

The file `gravity.list` is ended by a  `Binary file /etc/pihole/list.preEventHorizon matches` errror.

**Steps to reproduce:**

Currently I have the following host-files added:
```
https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts
http://localhost/adblock.hosts
http://mirror1.malwaredomains.com/files/justdomains
http://sysctl.org/cameleon/hosts
https://zeustracker.abuse.ch/blocklist.php?download=domainblocklist
https://s3.amazonaws.com/lists.disconnect.me/simple_tracking.txt
https://s3.amazonaws.com/lists.disconnect.me/simple_ad.txt
https://hosts-file.net/ad_servers.txt
https://raw.githubusercontent.com/crazy-max/WindowsSpyBlocker/master/data/hosts/win10/spy.txt
http://securemecca.com/Downloads/hosts.txt
https://raw.githubusercontent.com/quidsup/notrack/master/trackers.txt
https://raw.githubusercontent.com/Dawsey21/Lists/master/main-blacklist.txt
https://raw.github.com/notracking/hosts-blocklists/master/hostnames.txt
https://media.kuketz.de/lists/mobile_list.txt
https://www.netguard.me/hosts
https://hosts-file.net/download/hosts.txt
```
... and some special own lists on localhost. 
When I run pihole -g the error occurs.

**Troubleshooting undertaken, and/or other relevant information:**

Adding` -a` to grep at line 452 in gravity.sh helps me:
```
> diff gravity.sh gravity.sh.bak 

452c452
<   grep -F -a -x -v -f "${whitelistFile}" "${piholeDir}/${preEventHorizon}" > "${piholeDir}/${whitelistMatter}"
---
>   grep -F -x -v -f "${whitelistFile}" "${piholeDir}/${preEventHorizon}" > "${piholeDir}/${whitelistMatter}"
```

**Comment by StarPicard** [comment]
#816

**Comment by PromoFaux** [comment]
It's this file:
`https://hosts-file.net/download/hosts.txt` 

Interestingly, If I download it on my desktop, the file is `1251kb`.. watch what happens when I download it on my pi-hole device:

```
wget https://hosts-file.net/download/hosts.txt ~/hosts.txt
--2018-01-13 14:52:10--  https://hosts-file.net/download/hosts.txt
Resolving hosts-file.net (hosts-file.net)... 107.22.171.143
Connecting to hosts-file.net (hosts-file.net)|107.22.171.143|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 30496912 (29M) [text/plain]
Saving to: ‘hosts.txt’

hosts.txt                                                  100%[=====================================================================================================================================>]  29.08M   580KB/s    in 77s

2018-01-13 14:53:29 (385 KB/s) - ‘hosts.txt’ saved [30496912/30496912]

/home/fap/hosts.txt: Scheme missing.
FINISHED --2018-01-13 14:53:29--
Total wall clock time: 1m 19s
Downloaded: 1 files, 29M in 1m 17s (385 KB/s)
```

**Comment by PromoFaux** [comment]
wait... ignore this, I downloaded the wrong list on my desktop! .. need coffee

**Comment by PromoFaux** [comment]
Doing some more playing, I've downloaded the file on it's own and added only that to my adlists.list with `file:///path/to/hosts.txt` Just placing this here to collect thoughts

```
fap@fappotron:~$ pihole -g
  [i] Neutrino emissions detected...
  [✓] Pulling blocklist source list into range

  [i] Target: local (hosts.txt)
  [✓] Status: Retrieval successful

  [✓] Consolidating blocklists
  [✓] Extracting domains from blocklists
  [i] Number of domains being pulled in by gravity: 883,777
  [✓] Removing duplicate domains
  [i] Number of unique domains trapped in the Event Horizon: 883,777

  [i] Number of blocklist source domains being added to the whitelist: 1
  [i] Number of whitelisted domains: 14
  [i] Number of blacklisted domains: 20
  [✓] Parsing domains into hosts format
  [✓] Cleaning up stray matter

  [✓] Force-reloading DNS service
  [✓] DNS service is running
  [✓] Pi-hole blocking is Enabled
fap@fappotron:~$ tail /etc/pihole/gravity.list
192.168.0.2 zzzxxxcc.no-ip.biz
2a02:390:7bed:0:3aea:a7ff:fea1:646 zzzxxxcc.no-ip.biz
192.168.0.2 zzzzc.homedrugseshop.ru
2a02:390:7bed:0:3aea:a7ff:fea1:646 zzzzc.homedrugseshop.ru
192.168.0.2 zzzzz.cauction.hop.clickbank.net
2a02:390:7bed:0:3aea:a7ff:fea1:646 zzzzz.cauction.hop.clickbank.net
192.168.0.2 zzzzz.regmum.hop.clickbank.net
2a02:390:7bed:0:3aea:a7ff:fea1:646 zzzzz.regmum.hop.clickbank.net
192.168.0.2 Binary file /etc/pihole/list.preEventHorizon matches
2a02:390:7bed:0:3aea:a7ff:fea1:646 Binary file /etc/pihole/list.preEventHorizon matches
```

**Comment by PromoFaux** [comment]
Got it. Took a lot of comparing files to work it out!

That `hosts.txt` file contains this domain `www.hualañe.cl`, one would assume it's the `ñ` char throwing it off...

**Comment by davekr8** [comment]
Duplicate of #1862

**Comment by StarPicard** [comment]
I think so, as I mentioned adding `-a` to grep at line 452 in gravity.sh helps me.
This switch means that the file is treated as a text file if I understood it correctly.

**Comment by Mcat12** [comment]
@StarPicard Does #1910 fix this for you?

**Comment by StarPicard** [comment]
Have checked it manually on my pi and it seems to solve the problem.
I would recommend replacing [line 333](https://github.com/pi-hole/pi-hole/blob/876170767c72d3db9b319214593d0a8921773a78/gravity.sh#L333) as well.

**Comment by jacobsalmela** [comment]
This was released in v3.3.
